src/webots/vrml/WbNode.cpp

```diff
@@ -1116,6 +1120,24 @@ QStringList WbNode::listTextureFiles() const {
   return list;
 }
 
+const WbNode *WbNode::containingProto(bool skipThis) const {
+  const WbNode *n = this;
+  while (n) {
+    const WbProtoModel *protoModel = n->proto();
+    if (protoModel && (!skipThis || (skipThis && n != this)))
+      return n;
```

Comment:
```suggestion
    if (protoModel && (!skipThis || n != this))
      return n;
```

`n != this` will be evaluated only if `skipThis` is `true`

Reply:
I don't think I understand, or at least, what I understood is intentional.
Basically the idea is: if I pass a node n to this function, and let's assume this node happens to be a PROTO already:
- if skipThis is false: I can return n immediately since it's already a PROTO (the second part of the OR isn't considered since the first is already TRUE)
- if skipThis is true: then I return n only if it differs (i.e. it would fail the check for n, but not for an upper PROTO of n)

Reply:
ah nevermind, I see what you mean

Reply:
It simplifies the expression, but I don't know if it is maybe clearer to let the `(skipThis &&`